### ISSUE TITLE
Copy the file and paste into the same folder

### ISSUE THREAD

**Issue by dreamjade** [jacobwhall/marktext#20]
### Describe your feature request:

I would like to be able to copy a file and paste it into the same folder in MarkText. This would be useful for creating new files from existing ones, or for making copies of files to edit without modifying the originals.
Like the "duplicate" method in the jupyter notebook

### What problem does this feature solve?

Currently, the only way to copy and paste a file in MarkText is to copy it to the clipboard and then create a new file in the desired folder and paste the contents of the clipboard into it. This is a two-step process that can be tedious if you need to copy and paste multiple files.

### Additional context:

* Here are some possible ways to implement this feature:
Allow users to right-click on a file in MarkText and then drag it to the desired location in the same folder to create a copy of the file.
I hope you will consider adding this feature to MarkText. It would be a very useful addition for many users.

* Additionally:
To rename the newly created file, you could add a dialog box that pops up after the file is copied and pasted, asking the user for a new name for the file.

**Comment by jacobwhall** [comment]
Yes, I like this idea a lot. I have many markdown templates I use from time to time, and the workflow for duplicating them within MarkText should be easy. In my view the "duplicate" action should be separate from "rename" or "move", but I welcome further discussion.

I may not get to this for a while. I welcome any PRs.
